lib/reversetunnel/agentpool.go

```diff
@@ -107,10 +107,16 @@ func (m *AgentPool) FetchAndSyncAgents() error {
 func (m *AgentPool) pollAndSyncAgents() {
 	ticker := time.NewTicker(defaults.ReverseTunnelsRefreshPeriod)
 	defer ticker.Stop()
+	m.FetchAndSyncAgents()
 	for {
 		select {
 		case <-m.closeBroadcast.C:
 			m.Infof("closing")
+			m.Lock()
+			defer m.Unlock()
```

Comment:
AFAIK this introduces deadlock. as deferred function will be executed once function returns,not at the scope return, this means that any subsequent calls will block forever